tracee-ebpf/tracee/proc.go

```diff
@@ -0,0 +1,50 @@
+package tracee
+
+import (
+	"github.com/prometheus/procfs"
+)
+
+// gatherProcessTreeMap will put all PIDs into the returned map that are
+// children of filteredPID
+func gatherProcessTreeMap(filteredPID uint32) (map[uint32]uint32, error) {
+	fs, err := procfs.NewFS("/proc")
+	if err != nil {
+		return nil, err
+	}
+
+	procs, err := fs.AllProcs()
+	if err != nil {
+		return nil, err
+	}
+
+	filterPidAndChildPids := map[uint32]uint32{
+		filteredPID: 1,
+	}
+
+	processMap := map[uint32][]uint32{}
+	for i := range procs {
+		stat, err := procs[i].Stat()
+		if err != nil {
+			return nil, err
+		}
+		processMap[uint32(stat.PPID)] = append(processMap[uint32(stat.PPID)], uint32(stat.PID))
```

Comment:
I think that using this method you will only have the pids, and not the tids.
This means that the threads of a process tree won't be included, which is not what we want.

Reply:
This is still a problem...

Reply:
The process tree map in bpf code is using tgid and in userspace we use pid, aren't these the same?

Reply:
Yes, I see that the bpf code is now using tgid. Should be ok now, however I wonder if it wouldn't have been better to set tids both in userspace and and bpf code for better granularity of the filter. Probably not that important, just a thought.